research/integrations/fedora-ci/index.md

```diff
@@ -0,0 +1,338 @@
+
+There are premade tmt plans that we can run.
+
+:::note Question
+
+Should these be run in parallel or sequentially and fail on first?
+
+:::
+
+#### Installability test
+
+Installability test run by Fedora CI triggers Testing Farm and uses the
+following tmt plan: https://github.com/fedora-ci/installability-pipeline
+
+Installability test run by Zuul CI is done in the Zuul itself.
+
+### Testing Farm
+
+Different form of request than for upstream jobs.
+
+```json
+{
+  "id": "2fdee1a0-0746-45ad-ad0c-d3ec8833630b",
+  "user_id": "c4af7afe-b95e-4cf1-b989-9c458f0eecf0",
+  "token_id": "c4af7afe-b95e-4cf1-b989-9c458f0eecf0",
+  "test": {
+    "fmf": {
+      "url": "https://src.fedoraproject.org/forks/packit-stg/rpms/packit",
+      "ref": "0d4dfcfcc2a708f06c65b80404cc022e7bf6bead",
+      "merge_sha": null,
+      "path": ".",
+      "name": null,
+      "settings": null,
+      "plan_filter": null,
+      "test_name": null,
+      "test_filter": null
+    },
+    "script": null,
+    "sti": null
+  },
+  "state": "complete",
+  "environments_requested": [
+    {
+      "arch": "x86_64",
+      "os": {
+        "compose": "CentOS-Stream-9"
+      },
+      "pool": null,
+      "variables": {
+        "KOJI_TASK_ID": "123014869"
+      },
+      "secrets": null,
+      "artifacts": [
+        {
+          "id": "123014869",
+          "type": "fedora-koji-build",
+          "packages": null,
+          "install": true,
+          "order": 50
+        }
+      ],
+      "settings": null,
+      "tmt": {
+        "context": {
+          "arch": "x86_64",
+          "distro": "centos-stream-9",
+          "initiator": "fedora-ci",
+          "trigger": "commit"
+        },
+        "environment": null
+      },
+      "hardware": null,
+      "kickstart": null
+    }
+  ],
+  "notes": [
+    {
+      "level": "info",
+      "message": "tf-tmt/dispatch-1725615078-80d05f05"
+    }
+  ],
+  "result": {
+    "summary": null,
+    "overall": "passed",
+    "xunit": "…",
+    "xunit_url": "https://artifacts.dev.testing-farm.io/2fdee1a0-0746-45ad-ad0c-d3ec8833630b/results.xml"
+  },
+  "run": {
+    "console": null,
+    "stages": null,
+    "artifacts": "https://artifacts.dev.testing-farm.io/2fdee1a0-0746-45ad-ad0c-d3ec8833630b"
+  },
+  "settings": null,
+  "user": {
+    "webpage": null
+  },
+  "queued_time": 6.892783,
+  "run_time": 1207.608357,
+  "created": "2024-09-06T09:31:17.471815",
+  "updated": "2024-09-06T09:31:17.471844"
+}
+```
+
+Most notable differences:
+
+- RPMs are provided from Koji scratch build (following Fedora CI, which is
+  easier than self-hosting)
+- chroot is defined by the branch rather than config
+
+## Technical issues / questions
+
+### Load balancing
+
+Preferably it would be ideal to have separate deployment for the “dist-git only”
+service as handling all Fedora packages would definitely increase traffic and
+could cause issues with running as both Fedora CI and original Packit Service.
```

Comment:
Would this mean `dist-git CI` and the current Packit? Or, Packit upstream and Packit Fedora automation (including Koji, Bodhi and maybe also propose/pull)?

Reply:
I've been thinking about a way to split this in some reasonable matter :D Based on the way Packit operates, the most reasonable would be splitting the upstream / downstream parts, i.e.,

- Packit Service (handles upstream **and** `propose-downstream` (as it is triggered by upstream))
- “Packit as Fedora CI” (handles downstream **and** `pull-from-upstream`)

**However** I rant somewhere about the config in the dist-git and default config :D and this complicates things a lot :/ Maybe we could treat PRs in a special way for the _downstream PRs_ (Koji + TF).

---

feels like whack-a-mole…